examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb

```diff
@@ -0,0 +1,512 @@
+    "clean_sterbeliste=[]\n",
+    "\n",
+    "# we loop through each sentence\n",
+    "for sentence in sterbeliste:\n",
+    "    labels=[]\n",
+    "    sentence = re.sub(r\"<mark>\",\"<start> \",sentence)\n",
+    "    sentence = re.sub(r\"</mark>\",\" <end>\",sentence)\n",
+    "    sentence = re.sub(r\"\\(|\\)|/\",\"\",sentence)\n",
+    "    sentence = sentence.split(\" \")\n",
+    "    sentence = list(filter(None,sentence))\n",
+    "    if_loc= False\n",
+    "    for word in sentence:\n",
+    "        if word=='<start>':\n",
+    "            if_loc= True\n",
+    "            location = []\n",
+    "        elif word=='<end>':\n",
+    "            labels +=[get_location_id(location, locations)] * len(location)\n",
+    "            if_loc= False\n",
+    "        else:\n",
+    "            if if_loc:\n",
+    "                location.append(word)\n",
+    "            else:\n",
+    "                labels.append(None)\n",
+    "    samples.append(labels)\n",
+    "    sentence= [word for word in sentence if word !=\"<start>\" if word!=\"<end>\"]\n",
+    "    clean_sterbeliste.append(sentence)\n",
+    "\n",
+    "print(f\"Length of the sentence: {len(sentence)}\")\n",
+    "print(f\"Sample_tagging:{samples[0]}\")\n",
+    "print(f\"Number of place names: {len(locations)}\")\n",
+    "print(f\"=======================================\")\n",
+    "assert len(sentence) == len(labels)\n",
+    "#print(f\"Sentence and labeles: {[(word, label) for (word, label) in zip(sentence, labels)]}\")\n",
+    "print(f\"Cleaned Sentence: {clean_sterbeliste[0]}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## 3.Building training data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 118,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "'''\n",
+    "X Matrix\n",
+    "\n",
+    "Dimensions # sentences x # sentences (one column samples df, len(sample))\n",
+    "\n",
+    "'''\n",
+    "X = pd.DataFrame(clean_sterbeliste)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "**T Matrix**\n",
+    "\n",
+    "np.sparse matirx 1 colum 0, 1 colum 1, rows are keywords\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 117,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(6, 2)"
+      ]
+     },
+     "execution_count": 117,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "'''\n",
+    "T Matrix\n",
+    "Dimensions: # place names x # 2\n",
+    "\n",
+    "'''\n",
+    "import numpy as np\n",
+    "a= np.zeros(len(locations))\n",
+    "b= np.full(len(locations),1)\n",
+    "\n",
+    "T = np.stack((b,a.T))\n",
+    "T.T.shape\n"
```

Comment:
bad naming, better to change `T` to  e.g. `t_matrix` (and then `Z`to `z_matrix` for consistency reason)